Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -469,8 +469,60 @@ namespace OpenInfraPlatform {
 			void convertIfcOpenCrossProfileDef(const EXPRESSReference<typename IfcEntityTypesT::IfcOpenCrossProfileDef>& profileDef,
 				std::vector<std::vector<carve::geom::vector<2>>>& paths) const
 			{
-				throw oip::UnhandledException(profileDef);
+				if (profileDef->expired())
+					throw oip::ReferenceExpiredException(profileDef);
+
+				std::vector<carve::geom::vector<2>> outer_loop;
+
+				bool horizontal = profileDef->HorizontalWidths;
+
+				std::vector<double> widths;
+				widths.reserve(profileDef->Widths.size());
+				std::transform(
+					profileDef->Widths.begin(),
+					profileDef->Widths.end(),
+					widths.begin(),
+					[](auto& it) {return it * UnitConvert()->getLengthInMeterFactor(); });
+
+				std::vector<double> slopes;
+				slopes.reserve(profileDef->Slopes.size());
+				std::transform(
+					profileDef->Slopes.begin(),
+					profileDef->Slopes.end(),
+					slopes.begin(),
+					[](auto& it) {return it * UnitConvert()->getAngleInRadianFactor(); });

```

Comment:
This whole part is not necessary as widths[i] and slopes[i] should also be useable without converting them to std::vector

Reply:
I agree. I wanted @SamuilsRulovs to learn `std::transform`. 

But, yes, it is possible without the `vector`s and merely directly using `profileDef->Slopes[i] * UnitConvert()...` in the function call `calculate...`.

Reply:
Resolved: 5c1d3b6